modules/ismp/clients/parachain/client/src/lib.rs

```diff
@@ -254,3 +263,50 @@ impl<T: Config> RelayChainOracle for Pallet<T> {
 		RelayChainStateCommitments::<T>::get(height)
 	}
 }
+
+/// 6s slot or 12s slot duration based on current parablock progression by the relaychain mechanism
+#[derive(
+	Debug,
+	Clone,
+	Copy,
+	Encode,
+	Decode,
+	scale_info::TypeInfo,
+	MaxEncodedLen,
+	PartialEq,
+	Eq,
+	Hash,
+	Ord,
+	PartialOrd,
+	serde::Deserialize,
+	serde::Serialize,
+)]
+pub enum ParaSlotDuration {
+	/// synchronous backed with slot duration in milliseconds
+	Sync(u64),
+	/// asynchronous backed with slot duration in milliseconds
+	Async(u64),
+}
+/// Data provided when registering a parachain to be tracked by hyperbridge consensus client
+#[derive(
+	Debug,
+	Clone,
+	Copy,
+	Encode,
+	Decode,
+	scale_info::TypeInfo,
+	MaxEncodedLen,
+	PartialEq,
+	Eq,
+	Hash,
+	Ord,
+	PartialOrd,
+	serde::Deserialize,
+	serde::Serialize,
+)]
+pub struct ParachainData {
+	/// parachain id
+	pub id: u32,
+	/// parachain slot duration type
+	pub slot_duration_type: ParaSlotDuration,
```

Comment:
Slot duration should just be a u64 , this enum is not really doing anything special.

Reply:
This will help us even in the future to know which chains are using which parablock progressing mechanism

Reply:
That's not a big deal, we can always know from the slot duration.